cypress/integration/crd.spec.js

```diff
@@ -0,0 +1,177 @@
+
+    cy.get('[data-cy=delete]').click();
+    cy.get('[data-cy=confirm-delete]').click();
+    cy.wait(500);
+
+    cy.reload();
+    cy.get('[data-cy=zero-state]').should('be.visible');
+  });
+});
+
+describe('crd details page - non-existent crd', () => {
+  it('gives an error', () => {
+    cy.visit('/#/crd/foos.samplecontroller.k8s.io?namespace=default');
+    cy.wait(500);
+
+    cy.url().should('include', 'error');
+    cy.contains('404').should('exist');
+  });
+});
+
+describe('crd details page - valid crd (foo)', () => {
+  before(() => {
+    cy.fixture('crd.yaml').then(crd => {
+      cy.createResource({content: crd});
+    });
+  });
+
+  it('loads correctly', () => {
+    cy.visit('/#/crd/foos.samplecontroller.k8s.io?namespace=default');
+    cy.wait(500);
+
+    cy.get('kd-object-meta').within(() => {
+      cy.get('kd-property').first().contains('foos.samplecontroller.k8s.io');
+    });
+  });
+
+  it('has resource information', () => {
+    cy.get('[data-cy=resource-information]').within(() => {
+      cy.get('kd-property').should('have.length', 3);
+
+      cy.get('kd-property').eq(0).contains('v1alpha1');
+      cy.get('kd-property').eq(1).contains('Namespaced');
+      cy.get('kd-property').eq(2).contains('samplecontroller.k8s.io');
+    });
+  });
+
+  it('has accepted names', () => {
+    cy.get('[data-cy=accepted-names]').within(() => {
+      cy.get('kd-property').should('have.length', 4);
+
+      cy.get('kd-property').eq(0).contains('foos');
+      cy.get('kd-property').eq(1).contains('foo');
+      cy.get('kd-property').eq(2).contains('Foo');
+      cy.get('kd-property').eq(3).contains('FooList');
+    });
+  });
+
+  it('empty object section if no object exists', () => {
+    cy.get('kd-crd-object-list').within(() => {
+      cy.get('[data-cy=zero-state]').should('be.visible');
+    });
+  });
+
+  it('has one version', () => {
+    cy.get('kd-crd-versions-list').within(() => {
+      cy.get('mat-row').should('have.length', 1);
+
+      cy.get('mat-cell').first().contains('v1alpha1');
+    });
+  });
+
+  it('deletes crd from actionbar', () => {
+    cy.get('kd-actionbar-detail-delete').click();
+
+    cy.get('[data-cy=confirm-delete]').click();
+
+    cy.get('[data-cy=zero-state]').should('be.visible');
+  })
+});
+
+describe('crd objects page - non-existent object', () => {
+  it('should give an error', () => {
+    cy.visit('/#/crd/foos.samplecontroller.k8s.io/default/example-foo?namespace=default');
+
+    cy.url().should('include', 'error');
+    cy.contains('404').should('exist');
+  });
+});
+
+describe('crd objects list', () => {
+  before(() => {
+    cy.fixture('crd.yaml').then(crd => {
+      cy.createResource({content: crd})
+    });
+
+    cy.fixture('example-foo.yaml').then(object => {
+      cy.createResource({content: object, namespace: 'default'});
+    });
+  });
+
+  it('shows objects if they exist', () => {
+    cy.visit('/#/crd/foos.samplecontroller.k8s.io?namespace=default');
+    cy.wait(500);
```

Comment:
Try to avoid this type of `cy.wait` at all cost. It's not reliable. You can use `cy.route` and `cy.wait(alias)` to wait for actual request from the server.

Reply:
Can you change that?

Reply:
Yes, I'll fix this soon.

Reply:
Any progress?

Reply:
Yes. This is done.